fili-core/src/main/java/com/yahoo/bard/webservice/logging/TimedPhase.java

```diff
@@ -0,0 +1,79 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.logging;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Represents a phase that is timed.
+ * TimedPhase is used to associate a Timer located in the registry with the exact duration of such a phase for a
+ * specific request.
+ */
+public class TimedPhase implements AutoCloseable {
+    private static final Logger LOG = LoggerFactory.getLogger(TimedPhase.class);
+
+    private final String name;
+    private long start;
+    private long duration;
+
+    /**
+     * Constructor.
+     * Times are in nanoseconds.
+     *
+     * @param name  Name of the phase
+     */
+    public TimedPhase(String name) {
+        this.name = name;
+    }
+
+    /**
+     * Start the phase.
+     *
+     * @return This phase after being started
+     */
+    public TimedPhase start() {
+        if (isStarted()) {
+            LOG.warn("Tried to start timer that is already running: {}", name);
+        } else {
+            start = System.nanoTime();
+        }
+        return this;
+    }
+
+    /**
+     * Stop the phase.
+     */
+    public void stop() {
+        if (!isStarted()) {
+            LOG.warn("Tried to stop timer that has not been started: {}", name);
+            return;
+        }
+        duration += System.nanoTime() - start;
+        start = 0;
+    }
+
+    public long getDuration() {
+        return duration;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public TimeUnit getUnit() {
+        return TimeUnit.NANOSECONDS;
+    }
+
+    public boolean isStarted() {
```

Comment:
This may seem stupid, but should we change the name of this method to `isRunning`? We seem to be using it to indicate if this `TimedPhase` is currently _running_ or not, and `isStarted` feels more like a "once I've been started, I've always been started" sort of thing, where as running / stopped feels more like a toggleable state.

Obviously this is a small thing, but every time I see `isStarted`, I noticed I had to mentally pause and convert to `isRunning`.